### PR TITLE
cdmInstance may not be available when webKitMediaOpenCDMDecryptorRese…

### PR DESCRIPTION
**Pull request by goruklu**
…tSessionFromKeyIdIfNeeded() is called

This prevents a crash seen when playing content in http://www.gurdal-lab.net/dash.js/samples/playreadyMS.html

Signed-off-by: Gurdal Oruklu <gurdal_oruklu@comcast.com>

### CODE REVIEW

File: Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp [hunk 1]
```diff
@@ -153,6 +153,12 @@ static SessionResult webKitMediaOpenCDMDecryptorResetSessionFromKeyIdIfNeeded(We
     LockHolder locker(priv->m_mutex);
     RefPtr<WebCore::CDMInstance> cdmInstance = webKitMediaCommonEncryptionDecryptCDMInstance(self);
     ASSERT(cdmInstance && is<WebCore::CDMInstanceOpenCDM>(*cdmInstance));
+
+    if(!cdmInstance){
```
Comment (calvaris) [inline Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp]: There is an assert before this, so please, let's adapt this into `if (!cdmInstance || !is<WebCore::CDMInstanceOpenCDM>(*cdmInstance))`. Then remove the assert before and move it to inside the if in form of `ASSERT_NOT_REACHED()`.

**Review by calvaris — CHANGES_REQUESTED** [review]

File: Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp (hunk 1 above)
Comment (eocanha) [reply Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp]: Before accepting this patch or its corrected version, I'm trying to understand why the CDMInstance isn't available on time.

**Comment by goruklu** [comment]
@calvaris I've updated the ASSERT() as you requested. Let me know if this looks OK to you. I'm still seeing this crash even after the change which locks CDM key updates.

File: Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp [hunk 2]
```diff
@@ -152,9 +152,9 @@ static SessionResult webKitMediaOpenCDMDecryptorResetSessionFromKeyIdIfNeeded(We
 
     LockHolder locker(priv->m_mutex);
     RefPtr<WebCore::CDMInstance> cdmInstance = webKitMediaCommonEncryptionDecryptCDMInstance(self);
-    ASSERT(cdmInstance && is<WebCore::CDMInstanceOpenCDM>(*cdmInstance));
 
-    if(!cdmInstance){
+    if( !cdmInstance || !is<WebCore::CDMInstanceOpenCDM>(*cdmInstance) )
+        ASSERT_NOT_REACHED();
```
Comment (calvaris) [inline Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp:157]: We should ASSERT always after gstreamer logging, that way the log would be the last thing we see before the crash in debug mode.

**Review by calvaris — CHANGES_REQUESTED** [review]
Besides this you need to remove my commit from this PR. My commit has still to be investigated and it corresponds to fixing or mitigating another problem, not this one.

This one is weird. It looks like the app is not attanching the MediaKeys in time and we need to know why.

**Comment by goruklu** [comment]
replaced with #688

**Pull request closed without merging**